packages/gas-fee-controller/src/GasFeeController.ts

```diff
@@ -319,48 +317,30 @@ export class GasFeeController extends BaseControllerV2<
     this.pollTokens = new Set();
     this.getCurrentNetworkEIP1559Compatibility =
       getCurrentNetworkEIP1559Compatibility;
-
     this.getCurrentNetworkLegacyGasAPICompatibility =
       getCurrentNetworkLegacyGasAPICompatibility;
-
     this.getCurrentAccountEIP1559Compatibility =
       getCurrentAccountEIP1559Compatibility;
+    this.#getProvider = getProvider;
     this.EIP1559APIEndpoint = EIP1559APIEndpoint;
     this.legacyAPIEndpoint = legacyAPIEndpoint;
     this.clientId = clientId;
+
+    this.ethQuery = new EthQuery(this.#getProvider());
+
     if (onNetworkStateChange && getChainId) {
-      const initialProvider = getProvider();
-      this.ethQuery = new EthQuery(initialProvider);
       this.currentChainId = getChainId();
-      onNetworkStateChange(async () => {
-        const newProvider = getProvider();
-        const newChainId = getChainId();
-        this.ethQuery = new EthQuery(newProvider);
-        if (this.currentChainId !== newChainId) {
-          this.currentChainId = newChainId;
-          await this.resetPolling();
-        }
+      onNetworkStateChange(async (networkControllerState) => {
```

Comment:
Just noticed this breaking change that we missed. The state parameter wasn't used before, but now it is.

Reply:
Ah. That's a good point.

Reply:
Changelog updated here: https://github.com/MetaMask/core/pull/1405